core/modules/layout/layout.admin.inc

```diff
@@ -2645,3 +2648,96 @@ function layout_configure_region_page_submit($form, &$form_state) {
+
+  $module_blocks = layout_get_block_info();
+  $usage = layout_get_block_usage();
+
+  // Get all blocks.
+  foreach ($module_blocks as $module => $blocks) {
+    if (!empty($blocks)) {
+      foreach ($blocks as $delta => $block) {
+        $contexts = !empty($block['required contexts']) ? implode(",", $block['required contexts']) : '';
+        $usages = array();
+        if (isset($usage[$delta])) {
+          foreach ($usage[$delta] as $layout => $layout_use) {
+            $instances = array();
+            foreach ($layout_use as $position => $block_use) {
+              foreach ($block_use as $instance) {
+                $instances[$position][] = $instance;
+              }
+            }
+            $positions = implode(', ', array_map('ucfirst', array_keys($instances)));
+            $usages[] = l($instance->layout_title . ' (' . $positions . ')', 'admin/structure/layouts/manage/' . $layout);
+          }
+        }
+        $usage_list = $usages ? theme('item_list', array('items' => $usages)) : t('Not in use');
+        $block_name = $block['info'] . ' (' . $delta . ')';
+        $rows[] = array('name' => $block_name, 'usage' => $usage_list, 'module' => $module, 'contexts' => $contexts);
+      }
+    }
+  }
+
+  $header = array(
+    array('data' => 'Name', 'field' => 'name', 'sort' => 'ASC'),
+    array('data' => 'Usage', 'field' => 'usage', 'sort' => 'ASC'),
+    array('data' => 'Module', 'field' => 'module', 'sort' => 'ASC'),
+    array('data' => 'Contexts', 'field' => 'contexts'),
+  );
+
+  $order = tablesort_get_order($header);
+  $sort = tablesort_get_sort($header);
+
+  backdrop_sort($rows, array($order['sql'] => SORT_STRING));
+
+  $page['description_container'] = array(
+    '#type' => 'container',
+  );
+
+  $page['description_container']['description'] = array(
+    '#markup' => t("Provides a summary of all available blocks. The 'Usage' column shows which layout and region to which blocks are currently assigned in the format 'Layout name (region)'."),
+  );
+
+  $page['table'] = array(
+    '#theme' => 'table',
+    '#header' => $header,
+    '#rows' => $rows,
+    '#empty' => t('There are no blocks.'),
+  );
+
+  return $page;
+}
+
+/**
+ * Lists information about blocks in use in layouts.
+ *
+ * @param string $delta
+ *   The block delta.
+ *
+ * @return array
+ *   An array of block usage instances keyed by block delta, layout 
+ *   machine_name, and position. Each instance is a block object with an
+ *   additional key for the layout name.
+ */
+function layout_get_block_usage($delta = NULL) {
+  $blocks = &backdrop_static(__FUNCTION__, array());
+  if (empty($blocks)) {
+    $layout_info = layout_load_all();
+    foreach ($layout_info as $layout) {
+      foreach ($layout->positions as $position => $uuids) {
+        foreach ($uuids as $uuid) {
+          $block = $layout->content[$uuid];
+          $block->layout_title = $layout->title;
+          $blocks[$block->delta][$layout->name][$position][] = $block;
```

Comment:
`$block->delta` is not unique. I could create a custom block with the delta "main_menu" for example and it would conflict with System module's "main_menu" block. As such this array needs to be nested another level I think. e.g. `$blocks[$block->module][$block->delta][$layout->name][$position]` (phew!)